### PR TITLE
catch JSON parse error

### PR DESCRIPTION
**Pull request by jkvim**
Bug: when client pass invalid json argument to server, the server will break down. 

Fix: add `try...catch` block outside JSON parse.

when i test in curl with query as below

```
filter={where: {"userId": "hehe"}}
```

server will respond like below

```
{"error":{"name":"Error","status":400,"message":"Bad request: Invalid JSON.","statusCode":400}}
```

R= @yorkie  please review, thank you.

### CODE REVIEW

**Comment by jkvim** [comment]
@yorkie okay, i have test it by curl.

File: lib/application.js [hunk 1]
```diff
@@ -372,7 +372,14 @@ app.enableAuth = function(options) {
               ctx.args.filter = { fields: {} };
             }
             if (typeof ctx.args.filter === 'string') {
-              ctx.args.filter = JSON.parse(ctx.args.filter);
+              try {
+                ctx.args.filter = JSON.parse(ctx.args.filter);
+              } catch (error) {
+                var e = new Error('Bad request: Invalid JSON.');
```
Comment (yorkie) [inline lib/application.js]: Use `SyntaxError`, please.

File: lib/application.js [hunk 2]
```diff
@@ -372,7 +372,14 @@ app.enableAuth = function(options) {
...
+                e.statusCode = 400;
```
Comment (yorkie) [inline lib/application.js]: Set both `e.status` and `e.statusCode`.

**Comment by jkvim** [comment]
@yorkie thanks,  update that at https://github.com/weflex/neaty/pull/8/commits/2d6346e22801f2fe2e8998d457178ff406fd7267 . please review again.

File: lib/application.js [hunk 3]
```diff
@@ -372,7 +372,14 @@ app.enableAuth = function(options) {
               ctx.args.filter = { fields: {} };
             }
             if (typeof ctx.args.filter === 'string') {
-              ctx.args.filter = JSON.parse(ctx.args.filter);
+              try {
+                ctx.args.filter = JSON.parse(ctx.args.filter);
+              } catch (error) {
+                var e = new SyntaxError('Bad request: Invalid JSON.');
+                e.status = e.statusCode = 400;
```
Comment (yorkie) [inline lib/application.js]: We should use this https://httpstatuses.com/422

Comment (yorkie) [inline lib/application.js]: Sorry, 400 is exact perfect as https://httpstatuses.com/400

File: lib/application.js [hunk 4]
```diff
@@ -372,7 +372,14 @@ app.enableAuth = function(options) {
               ctx.args.filter = { fields: {} };
             }
             if (typeof ctx.args.filter === 'string') {
-              ctx.args.filter = JSON.parse(ctx.args.filter);
+              try {
+                ctx.args.filter = JSON.parse(ctx.args.filter);
+              } catch (error) {
+                var e = new SyntaxError('Bad request: Invalid JSON.');
```
Comment (yorkie) [inline lib/application.js]: Could you just write it as https://github.com/weflex/neaty/blob/fix/catch-json-parse/lib/application.js#L396-L413?

**Comment by jkvim** [comment]
@yorkie hi, i have update again https://github.com/weflex/neaty/pull/8/commits/2184817bfa76b151301032f7dcae82003b969236 , pleases review.

File: lib/application.js [hunk 5]
```diff
@@ -372,7 +391,16 @@ app.enableAuth = function(options) {
               ctx.args.filter = { fields: {} };
             }
             if (typeof ctx.args.filter === 'string') {
-              ctx.args.filter = JSON.parse(ctx.args.filter);
+              try {
+                ctx.args.filter = JSON.parse(ctx.args.filter);
+              } catch (error) {
+                errStatusCode = 400;
```
Comment (yorkie) [inline lib/application.js:397]: Where this did be declared?

File: lib/application.js [hunk 6]
```diff
@@ -372,7 +391,16 @@ app.enableAuth = function(options) {
...
+                var e = new SyntaxError(messages[errStatusCode].message + ' Invalid JSON');
```
Comment (yorkie) [inline lib/application.js]: Just use message is enough IMO

File: lib/application.js (hunk 5 above)
Comment (jkvim) [inline lib/application.js:397]: @yorkie  Line355 https://github.com/weflex/neaty/pull/8/files#diff-5372f626ee15242f1e2c6eb31655b4faR355

**Comment by jkvim** [comment]
@yorkie  fresh update https://github.com/weflex/neaty/pull/8/commits/42b28a452e13ad657ccbd15e9143afd1867f86d4

**Comment by yorkie** [comment]
@seanbot bump patch

**Comment by yorkie** [comment]
Landed with rewording title, thank you @jkvim

**Comment by jkvim** [comment]
@yorkie appreciate for your review 🍊 .

**Comment by yorkie** [comment]
:-)

**Pull request merged**